### PR TITLE
Edge Support

### PR DESCRIPTION
**Pull request by BenMcGarry**
Initial Edge Support

Due to Edge being preview, it is required to set special permissions onto the folder that the extension is in. This is done by running the following in command prompt from the extensions directory

`icacls "%cd%" /grant "*S-1-15-2-3624051433-2125758914-1423191267-1740899205-1073925389-3782572162-737981194":"(OI)(CI)(WDAC,WO,GE)"`

### CODE REVIEW

**Comment by BenMcGarry** [comment]
Current Open Issues: https://github.com/honestbleeps/Reddit-Enhancement-Suite/issues?q=is%3Aissue+is%3Aopen+label%3ABrowser-Edge

File: edge/background.js [hunk 1]
```diff
@@ -0,0 +1,327 @@
...
+	You should have received a copy of the GNU General Public License
+	along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+*/
+
+/* global chrome: false, msBrowser: false */
+
+/* Edge Support - PREVIEW */
+
+if (typeof msBrowser !== 'undefined') {
```
Comment (andytuba) [inline edge/background.js]: If this is the only change, maybe we could move `BrowserDetect` into a separate file and load it in `background.html`, then `if browser is edge) { chrome alias; }`.

File: edge/browsersupport-chrome.js [hunk 2]
```diff
@@ -0,0 +1,243 @@
+/* global chrome: false, msBrowser: false */
+
+/* Edge Support - PREVIEW */
+
+if (typeof msBrowser !== 'undefined') {
```
Comment (andytuba) [inline edge/browsersupport-chrome.js]: Same thing here -- make sure RESenvironment is loaded before browsersupport-chrome.js, then refactor this shim into `var chrome = RESenvironment.navigatorApiObject();` .. this idea needs a little polish.

File: edge/manifest.json [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+		"https://reddit.com/*",
+		"http://*.reddit.com/*",
+		"https://*.reddit.com/*",
+		"http://min.us/api/*",
+		"http://*.flickr.com/photos/*",
+		"http://img.photobucket.com/*",
+		"http://noembed.com/embed?url=*"
+	],
+	"optional_permissions": [
+		"https://api.twitter.com/*",
```
Comment (andytuba) [inline edge/manifest.json]: We should probably move all of these into `"permissions"`.

**Comment by andytuba** [comment]
Any changes to `gulpfile.js` needed for Node?

**Comment by BenMcGarry** [comment]
Run INSTALL_SCRIPT.cmd to give permissions to the folder

'use strict'; MUST be commented out after build in browsersupport-chrome.js otherwise the extension will not load.

Extension will load but issues [here](https://github.com/honestbleeps/Reddit-Enhancement-Suite/issues?q=is%3Aissue+is%3Aopen+label%3ABrowser-Edge) are still present.

**Comment by BenMcGarry** [comment]
Full edge support committed, pulls needed files from chrome and only required are in edge directory.

Current known issues: https://github.com/honestbleeps/Reddit-Enhancement-Suite/issues?q=is%3Aissue+is%3Aopen+label%3ABrowser-Edge

**Pull request closed without merging**